Workload files when specified with its full path don't work correctly

**Description of the bug**
Workload files containing scenario(s) when specified with its full path doesn't work. This is with [nb version `3.12.124`](https://github.com/nosqlbench/nosqlbench/releases/tag/nosqlbench-3.12.134).

**To Reproduce**
Steps to reproduce the behavior:
1. Copy the contents from below and create a new `ex02.yaml` file in a `~/Downloads` directory. FWIW, I'm using macOS Catalina and I did place this on my `Downloads` directory. Since markdown doesn't allow uploading of `.yaml` extension file. Please capture the below and create the workload file yourself.

<details>
  <summary>Click to expand and view the contents of the workload file</summary>

```
description: |
   This is an experiment

scenarios:
  default:
    ddl: run driver=stdout tags=phase:ddl cycles=UNDEF threads=auto
    load: run driver=stdout tags=phase:load cycles=UNDEF threads=auto


blocks:
  - tags:
      phase: ddl
    params:
      prepared: false
      ratio: 1
    statements:
      - p1s1: |
          Got here 1
        tags: 
          name: ddl_s1
      - p1s2: | 
          Got here 2
          Got here 3
        tags: 
          name: ddl_s2
  - tags:
      phase: load
    params:
      prepared: false
      ratio: 1
    statements:
      - p2s1: |
          Got here 4
        tags: 
           name: load_s1
```
</details>

2. File permissions are as follows,
```
$ pwd /Users/first.last/Downloads/ex02.yaml
-rw-r--r--@ 1 first.last  staff  684 Aug  7 14:39 ex02.yaml
```
3. Kick off the following command which produces a runtime exception,
```
$ java -jar nb.jar /Users/first.last/Downloads/ex02.yaml default -v --show-stacktraces
```
and the full stack of the exception is as follows,

<details>
  <summary>Click to expand</summary>

  ```
$ cd /Users/first.last/Desktop
$ java -jar nb.jar /Users/first.last/Downloads/ex02.yaml default -v --show-stacktraces
13:15:01.328 [main] INFO  io.nosqlbench.engine.cli.NBCLI - console logging level is INFO
13:15:01.348 [main] WARN  i.nosqlbench.engine.cli.NBCLIOptions - Console is already logging info or more, so progress data on console is suppressed.
13:15:01.354 [main] INFO  io.nosqlbench.engine.cli.NBCLI - Charting disabled
Logging to logs/scenario_20200807_131501_326.log
13:15:01.385 [scenarios:001] INFO  i.n.engine.core.script.Scenario - Using engine Graalvm
13:15:02.332 [scenarios:001] INFO  i.n.e.api.util.SimpleServiceLoader - Loaded Types:[stdout, diag, jmx, kafkaproducer, tcpserver, webdriver, http, cqlverify, mongodb, tcpclient, cql]
13:15:02.334 [scenarios:001] INFO  i.n.e.api.util.SimpleServiceLoader - Loaded Types:[stdout, diag, jmx, kafkaproducer, tcpserver, webdriver, http, cqlverify, mongodb, tcpclient, cql]
13:15:02.339 [scenarios:001] INFO  i.n.e.api.activityimpl.ParameterMap - getting entry set for (0)/{threads=auto, workload=Users/first.last/Downloads/ex02.yaml, cycles=3, tags=phase:ddl, driver=stdout, alias=ex02_default_000}
13:15:02.402 [main] INFO  i.n.e.core.script.ScenariosExecutor - scenarios executor shutdownActivity after 0ms.
13:15:02.403 [main] INFO  i.n.engine.core.ScenariosResults - results for scenario: name:'scenario_20200807_131501_326'
13:15:02.403 [main] INFO  i.n.engine.core.ScenarioResult - -- BEGIN METRICS DETAIL --
13:15:02.410 [main] INFO  i.n.engine.core.ScenarioResult - -- END METRICS DETAIL --
13:15:02.411 [main] INFO  io.nosqlbench.engine.cli.NBCLI - 1 scenarios, 0 normal, 1 errored
13:15:02.414 [main] ERROR i.n.engine.core.ScenarioErrorHandler - Error from driver or included library: java.lang.RuntimeException: javax.script.ScriptException: org.graalvm.polyglot.PolyglotException: error while reading file Users/first.last/Downloads/ex02.yaml
java.util.concurrent.ExecutionException: java.lang.RuntimeException: javax.script.ScriptException: org.graalvm.polyglot.PolyglotException: error while reading file Users/first.last/Downloads/ex02.yaml
    at java.base/java.util.concurrent.FutureTask.report(FutureTask.java:122)
    at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:191)
    at io.nosqlbench.engine.core.script.ScenariosExecutor.getAsyncResultStatus(ScenariosExecutor.java:152)
    at io.nosqlbench.engine.core.script.ScenariosExecutor.awaitAllResults(ScenariosExecutor.java:117)
    at io.nosqlbench.engine.core.script.ScenariosExecutor.awaitAllResults(ScenariosExecutor.java:75)
    at io.nosqlbench.engine.cli.NBCLI.run(NBCLI.java:317)
    at io.nosqlbench.engine.cli.NBCLI.main(NBCLI.java:53)
Caused by: java.lang.RuntimeException: javax.script.ScriptException: org.graalvm.polyglot.PolyglotException: error while reading file Users/first.last/Downloads/ex02.yaml
    at io.nosqlbench.engine.core.script.Scenario.run(Scenario.java:221)
    at io.nosqlbench.engine.core.script.Scenario.call(Scenario.java:234)
    at io.nosqlbench.engine.core.script.Scenario.call(Scenario.java:44)
    at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
    at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
    at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
    at java.base/java.lang.Thread.run(Thread.java:832)
Caused by: javax.script.ScriptException: org.graalvm.polyglot.PolyglotException: error while reading file Users/first.last/Downloads/ex02.yaml
    at com.oracle.truffle.js.scriptengine.GraalJSScriptEngine.eval(GraalJSScriptEngine.java:381)
    at com.oracle.truffle.js.scriptengine.GraalJSScriptEngine.access$400(GraalJSScriptEngine.java:77)
    at com.oracle.truffle.js.scriptengine.GraalJSScriptEngine$9.eval(GraalJSScriptEngine.java:533)
    at java.scripting/javax.script.CompiledScript.eval(CompiledScript.java:103)
    at io.nosqlbench.engine.core.script.Scenario.run(Scenario.java:206)
    ... 6 common frames omitted
Caused by: org.graalvm.polyglot.PolyglotException: error while reading file Users/first.last/Downloads/ex02.yaml
    at io.nosqlbench.engine.api.activityconfig.rawyaml.RawStmtsLoader.loadPath(RawStmtsLoader.java:47)
    at io.nosqlbench.engine.api.activityconfig.StatementsLoader.loadPath(StatementsLoader.java:91)
    at io.nosqlbench.activitytype.stdout.StdoutActivity.<init>(StdoutActivity.java:80)
    at io.nosqlbench.activitytype.stdout.StdoutActivityType.getActivity(StdoutActivityType.java:35)
    at io.nosqlbench.activitytype.stdout.StdoutActivityType.getActivity(StdoutActivityType.java:11)
    at io.nosqlbench.engine.api.activityapi.core.ActivityType.getAssembledActivity(ActivityType.java:73)
    at io.nosqlbench.engine.core.ScenarioController.getActivityExecutor(ScenarioController.java:268)
    at io.nosqlbench.engine.core.ScenarioController.run(ScenarioController.java:87)
    at io.nosqlbench.engine.core.ScenarioController.run(ScenarioController.java:77)
    at io.nosqlbench.engine.core.ScenarioController.run(ScenarioController.java:102)
    at io.nosqlbench.engine.core.PolyglotScenarioController.run(PolyglotScenarioController.java:41)
    at <js>.:program(<eval>:1)
    at org.graalvm.polyglot.Context.eval(Context.java:345)
    at com.oracle.truffle.js.scriptengine.GraalJSScriptEngine.eval(GraalJSScriptEngine.java:379)
    ... 10 common frames omitted
java.lang.RuntimeException: javax.script.ScriptException: org.graalvm.polyglot.PolyglotException: error while reading file Users/first.last/Downloads/ex02.yaml
  ```
</details>

**What was Expected**
I was expecting this to run and provide a successful output like how it is currently working when I navigate to the directory where the workload yaml file is and execute the following command,

```
$ cd /Users/first.last/Downloads/
$ java -jar nb.jar ./ex02.yaml default --progress console:1s --show-stacktraces
Logging to logs/scenario_20200807_131235_965.log
13:12:36.707 [scenarios:001] WARN  i.n.e.a.activityimpl.SimpleActivity - The stride does not evenly divide cycles. Only full strides will be executed,leaving some cycles unused.
Got here 1
Got here 2
Got here 3
```

**Additional context**
N/A

**Screenshots, if applicable**
N/A

I retested this by downloading the latest [`nb v3.12.135`](https://github.com/nosqlbench/nosqlbench/releases/download/nosqlbench-3.12.135/nb.jar) and it appears to be still error out with `java.lang.RuntimeException: javax.script.ScriptException: org.graalvm.polyglot.PolyglotException: error while reading file`. (FWIW, this release did indeed fix showing all 4 `Got here` output that was expected)

<details>
  <summary>Click here to expand and see the full stack trace</summary>

```
$ java -jar nb.jar --version
3.12.135

$ java -jar nb.jar /Users/first.last/Downloads/ex02.yaml default -v --show-stacktraces
09:51:12.592 [main] INFO  io.nosqlbench.engine.cli.NBCLI - console logging level is INFO
09:51:12.602 [main] WARN  i.nosqlbench.engine.cli.NBCLIOptions - Console is already logging info or more, so progress data on console is suppressed.
09:51:12.605 [main] INFO  io.nosqlbench.engine.cli.NBCLI - Charting disabled
Logging to logs/scenario_20200808_095112_590.log
09:51:12.637 [scenarios:001] INFO  i.n.engine.core.script.Scenario - Using engine Graalvm
09:51:13.576 [scenarios:001] INFO  i.n.e.api.util.SimpleServiceLoader - Loaded Types:[stdout, diag, jmx, kafkaproducer, tcpserver, webdriver, http, cqlverify, mongodb, tcpclient, cql]
09:51:13.578 [scenarios:001] INFO  i.n.e.api.util.SimpleServiceLoader - Loaded Types:[stdout, diag, jmx, kafkaproducer, tcpserver, webdriver, http, cqlverify, mongodb, tcpclient, cql]
09:51:13.583 [scenarios:001] INFO  i.n.e.api.activityimpl.ParameterMap - getting entry set for (1)/{driver=stdout, threads=auto, workload=Users/first.last/Downloads/ex02.yaml, alias=ex02_default_ddl, tags=phase:ddl}
09:51:13.661 [main] INFO  i.n.e.core.script.ScenariosExecutor - scenarios executor shutdownActivity after 0ms.
09:51:13.662 [main] INFO  i.n.engine.core.ScenariosResults - results for scenario: name:'scenario_20200808_095112_590'
09:51:13.663 [main] INFO  i.n.engine.core.ScenarioResult - -- BEGIN METRICS DETAIL --
09:51:13.669 [main] INFO  i.n.engine.core.ScenarioResult - -- END METRICS DETAIL --
09:51:13.671 [main] INFO  io.nosqlbench.engine.cli.NBCLI - 1 scenarios, 0 normal, 1 errored
09:51:13.673 [main] ERROR i.n.engine.core.ScenarioErrorHandler - Error from driver or included library: java.lang.RuntimeException: javax.script.ScriptException: org.graalvm.polyglot.PolyglotException: error while reading file Users/first.last/Downloads/ex02.yaml
java.util.concurrent.ExecutionException: java.lang.RuntimeException: javax.script.ScriptException: org.graalvm.polyglot.PolyglotException: error while reading file Users/first.last/Downloads/ex02.yaml
	at java.base/java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:191)
	at io.nosqlbench.engine.core.script.ScenariosExecutor.getAsyncResultStatus(ScenariosExecutor.java:152)
	at io.nosqlbench.engine.core.script.ScenariosExecutor.awaitAllResults(ScenariosExecutor.java:117)
	at io.nosqlbench.engine.core.script.ScenariosExecutor.awaitAllResults(ScenariosExecutor.java:75)
	at io.nosqlbench.engine.cli.NBCLI.run(NBCLI.java:312)
	at io.nosqlbench.engine.cli.NBCLI.main(NBCLI.java:48)
Caused by: java.lang.RuntimeException: javax.script.ScriptException: org.graalvm.polyglot.PolyglotException: error while reading file Users/first.last/Downloads/ex02.yaml
	at io.nosqlbench.engine.core.script.Scenario.run(Scenario.java:252)
	at io.nosqlbench.engine.core.script.Scenario.call(Scenario.java:265)
	at io.nosqlbench.engine.core.script.Scenario.call(Scenario.java:47)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
	at java.base/java.lang.Thread.run(Thread.java:832)
Caused by: javax.script.ScriptException: org.graalvm.polyglot.PolyglotException: error while reading file Users/first.last/Downloads/ex02.yaml
	at com.oracle.truffle.js.scriptengine.GraalJSScriptEngine.eval(GraalJSScriptEngine.java:381)
	at com.oracle.truffle.js.scriptengine.GraalJSScriptEngine.eval(GraalJSScriptEngine.java:343)
	at java.scripting/javax.script.AbstractScriptEngine.eval(AbstractScriptEngine.java:249)
	at io.nosqlbench.engine.core.script.Scenario.run(Scenario.java:235)
	... 6 common frames omitted
Caused by: org.graalvm.polyglot.PolyglotException: error while reading file Users/first.last/Downloads/ex02.yaml
	at io.nosqlbench.engine.api.activityconfig.rawyaml.RawStmtsLoader.loadPath(RawStmtsLoader.java:47)
	at io.nosqlbench.engine.api.activityconfig.StatementsLoader.loadPath(StatementsLoader.java:91)
	at io.nosqlbench.activitytype.stdout.StdoutActivity.<init>(StdoutActivity.java:77)
	at io.nosqlbench.activitytype.stdout.StdoutActivityType.getActivity(StdoutActivityType.java:35)
	at io.nosqlbench.activitytype.stdout.StdoutActivityType.getActivity(StdoutActivityType.java:11)
	at io.nosqlbench.engine.api.activityapi.core.ActivityType.getAssembledActivity(ActivityType.java:73)
	at io.nosqlbench.engine.core.ScenarioController.getActivityExecutor(ScenarioController.java:268)
	at io.nosqlbench.engine.core.ScenarioController.run(ScenarioController.java:87)
	at io.nosqlbench.engine.core.ScenarioController.run(ScenarioController.java:77)
	at io.nosqlbench.engine.core.ScenarioController.run(ScenarioController.java:102)
	at io.nosqlbench.engine.core.PolyglotScenarioController.run(PolyglotScenarioController.java:41)
	at <js>.:program(<eval>:1)
	at org.graalvm.polyglot.Context.eval(Context.java:345)
	at com.oracle.truffle.js.scriptengine.GraalJSScriptEngine.eval(GraalJSScriptEngine.java:379)
	... 9 common frames omitted
java.lang.RuntimeException: javax.script.ScriptException: org.graalvm.polyglot.PolyglotException: error while reading file Users/first.last/Downloads/ex02.yaml
```
</details>

--- a/nb-apis/nb-api/src/main/java/io/nosqlbench/nb/api/nbio/ResolverForFilesystem.java
+++ b/nb-apis/nb-api/src/main/java/io/nosqlbench/nb/api/nbio/ResolverForFilesystem.java
@@ -56,7 +56,15 @@ public class ResolverForFilesystem implements ContentResolver {
         if (uri.getScheme() != null && !uri.getScheme().isEmpty() && !uri.getScheme().equals("file")) {
             return null;
         }
-        Path pathFromUri = Path.of(uri.getPath());
+
+        String rawPath = uri.getPath();
+        
+        // Expand "~" to the user's home directory if it appears at the start of the path
+        if (rawPath.startsWith("~")) {
+            rawPath = System.getProperty("user.home") + rawPath.substring(1);
+        }
+        
+        Path pathFromUri = Path.of(rawPath);
 
         if (Files.isReadable(pathFromUri)) {
             return pathFromUri;

